### ISSUE TITLE
dockerfile_path doesn't work

### ISSUE THREAD

**Issue by meska** [jctaveras/heroku-deploy#8]
doesn't work if i set a dockerfile_path

<img width="1164" alt="immagine" src="https://user-images.githubusercontent.com/276923/92450011-2af37c80-f1bb-11ea-8f59-d61d2b7beeb1.png">

**Comment by jctaveras** [comment]
Hey @meska, Can you post the file structure of your project?

**Comment by meska** [comment]
sure, the folder I would like to to use as base for dockerfile is "backend".
tried both self hosted and git hosted.
tried with 'backend' , backend,  './backend/' with no success

<img width="517" alt="immagine" src="https://user-images.githubusercontent.com/276923/92485340-1929cd80-f1eb-11ea-95c5-8106e0ab7cf0.png">

action file:

<img width="475" alt="immagine" src="https://user-images.githubusercontent.com/276923/92485569-686ffe00-f1eb-11ea-925c-3522d2dcd508.png">

**Comment by jctaveras** [comment]
it's really weird, I was able to test this on a project that has the following structure:

![image](https://user-images.githubusercontent.com/29868501/92496694-77908500-f1c6-11ea-9ebf-08b73d0d02d7.png)

and this is the action configuration

```yml
name: Staging
on:
  pull_request:
    branches: [ staging ]
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v2
      - name: Build, Push and Deploy to Heroku
        id: heroku
        uses: jctaveras/heroku-deploy@v2.0.0
        with:
          email: ${{ secrets.HEROKU_EMAIL }}
          api_key: ${{ secrets.HEROKU_API_KEY }}
          app_name: ${{ secrets.HEROKU_APP_NAME }}
          dockerfile_path: 'server'
```

**Comment by meska** [comment]
another hint: If I put a non existing directory i have this  error:

<img width="638" alt="immagine" src="https://user-images.githubusercontent.com/276923/92499307-01a71080-f1fc-11ea-96be-ee78d722f66c.png">


probably you need to concatenate this:
`await exec('docker build . --file Dockerfile ${buildOptions} --tag registry.heroku.com/${appName}/web');`
to
`await exec('cd ${dockerFilePath}  && docker build . --file Dockerfile ${buildOptions} --tag registry.heroku.com/${appName}/web');`

**Comment by jctaveras** [comment]
Can you fork the repo and try the solution you are proposing? If it works, then you can open a PR

**Comment by greenphoenixeng** [comment]
I too am having the same problem. @jctaveras I followed your strategy in putting the Dockerfile in a folder, but still got an error. I was able to build the image locally. I tried `build` and `build/`

Error from workflow:
![image](https://user-images.githubusercontent.com/74696553/103592187-f74e0800-4eb7-11eb-8d8f-9f2a9d009cec.png)

Action config:
![image](https://user-images.githubusercontent.com/74696553/103592229-13ea4000-4eb8-11eb-9167-0dd7155734b6.png)

**Comment by mathix420** [comment]
Same issue

![image](https://user-images.githubusercontent.com/37625778/104166030-6d8fc480-53fa-11eb-8501-4f61d6584c42.png)

```yaml
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - name: Deploy Docker Image to Heroku App
        uses: jctaveras/heroku-deploy@v2.1.1
        with:
          email: ${{ secrets.HEROKU_EMAIL }}
          api_key: ${{ secrets.HEROKU_API_KEY }}
          app_name: ${{ secrets.HEROKU_APP_NAME }}
          dockerfile_path: image
```

**Comment by mathix420** [comment]
I manage to find how to fix the issue for me, I think it's the same as @greenphoenixeng and @meska .

Just needed to rewrite the code for it to run that
```bash
docker build ${dockerFilePath || '.'} ${buildOptions} --tag registry.heroku.com/${appName}/web'
```

**Comment by kacpercierpiak** [comment]
Hey, Still I have this issue..

**Comment by jctaveras** [comment]
> I manage to find how to fix the issue for me, I think it's the same as @greenphoenixeng and @meska .
> 
> Just needed to rewrite the code for it to run that
> 
> ```shell
> docker build ${dockerFilePath || '.'} ${buildOptions} --tag registry.heroku.com/${appName}/web'
> ```

Thanks, @mathix420 for this I was able to spot the issue and I applied the fix!

**Comment by jctaveras** [comment]
You can now use the version [v2.1.2](https://github.com/marketplace/actions/deploy-docker-image-to-heroku-app)

**Comment by jctaveras** [comment]
@meska @mathix420 @kacpercierpiak @greenphoenixeng I was able to pinpoint the issue and reproduced it myself I just made a quick fix on version [v2.1.3](https://github.com/marketplace/actions/deploy-docker-image-to-heroku-app) hope this fixes the issue

**Comment by budleigh** [comment]
Still broken in v.2.1.3. getting duplicates of the root directory, regardless of configuration:

```
unable to prepare context: unable to evaluate symlinks in Dockerfile path: lstat /home/runner/work/myrepo/myrepo/Dockerfile: no such file or directory
```

Same result with an empty string, no `dockerfile_path` at all, or ".".
